app/src/lib/stores/app-store.ts

```diff
@@ -3442,7 +3446,8 @@ export class AppStore extends TypedBaseStore<IAppState> {
 
       // If a local branch, user may have the branch to delete checked out and
       // we need to switch to a different branch (default or recent).
-      const branchToCheckout = this.getBranchToCheckoutAfterDelete(branch, r)
+      const branchToCheckout =
+        toCheckout || this.getBranchToCheckoutAfterDelete(branch, r)
```

Comment:
Nit: Should we use `??` here? I understand they should be equivalent in this specific case, but `||` does a falsy/truthy check which, in theory, we should avoid 🤔 
```suggestion
        toCheckout ?? this.getBranchToCheckoutAfterDelete(branch, r)
```

Reply:
Had to look that one up I had always used/seen `||` before - and yeah `??` makes more sense.